pkg/controller/whisker/whisker_controller.go

```diff
@@ -0,0 +1,279 @@
+	cli client.Client,
+	schema *runtime.Scheme,
+	statusMgr status.StatusManager,
+	p operatorv1.Provider,
+	tierWatchReady *utils.ReadyFlag,
+	opts options.AddOptions,
+) *Reconciler {
+	c := &Reconciler{
+		cli:            cli,
+		scheme:         schema,
+		provider:       p,
+		status:         statusMgr,
+		clusterDomain:  opts.ClusterDomain,
+		tierWatchReady: tierWatchReady,
+	}
+	c.status.Run(opts.ShutdownContext)
+	return c
+}
+
+// blank assignment to verify that ReconcileConnection implements reconcile.Reconciler
+var _ reconcile.Reconciler = &Reconciler{}
+
+// Reconciler reconciles a ManagementClusterConnection object
+type Reconciler struct {
+	cli                        client.Client
+	scheme                     *runtime.Scheme
+	provider                   operatorv1.Provider
+	status                     status.StatusManager
+	clusterDomain              string
+	tierWatchReady             *utils.ReadyFlag
+	resolvedPodProxies         []*httpproxy.Config
+	lastAvailabilityTransition metav1.Time
+}
+
+// Reconcile reads that state of the cluster for a Whisker object and makes changes based on the
+// state read and what is in the Whisker.Spec. The Controller will requeue the Request to be
+// processed again if the returned error is non-nil or Result.Requeue is true, otherwise upon completion it will
+// remove the work from the queue.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Whisker")
+	result := reconcile.Result{}
+
+	variant, installation, err := utils.GetInstallation(ctx, r.cli)
+	if err != nil {
+		return result, err
+	}
+
+	whiskerCR, err := utils.GetIfExists[operatorv1.Whisker](ctx, utils.DefaultInstanceKey, r.cli)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying Whisker CR", err, reqLogger)
+		return result, err
+	} else if whiskerCR == nil {
+		r.status.OnCRNotFound()
+		return result, nil
+	}
+	r.status.OnCRFound()
+	// SetMetaData in the TigeraStatus such as observedGenerations.
+	defer r.status.SetMetaData(&whiskerCR.ObjectMeta)
+
+	pullSecrets, err := utils.GetNetworkingPullSecrets(installation, r.cli)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error retrieving pull secrets", err, reqLogger)
+		return result, err
+	}
+
+	certificateManager, err := certificatemanager.Create(r.cli, installation, r.clusterDomain, common.OperatorNamespace())
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the certificate manager", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Copy the secret from the operator namespace to the guardian namespace if it is present.
+	tunnelSecret := &corev1.Secret{}
+	err = r.cli.Get(ctx, types.NamespacedName{Name: render.GuardianSecretName, Namespace: common.OperatorNamespace()}, tunnelSecret)
```

Comment:
Probably only need to do this if there's a management cluster connection present?

Reply:
Removed for now.